packages/store/addon/-private/system/record-identifier.ts

```diff
@@ -0,0 +1,40 @@
+import InternalModel from './model/internal-model';
+import { RecordData } from '..';
+export interface RecordIdentifier {
+    type: string;
+    id: string | null;
+    lid: string;
+}
+
+let typeIdMap = {};
```

Comment:
this will be an extremely (dangerously?) inferred type. Suggest at least
```suggestion
let typeIdMap: object = {};
```